great_expectations/data_context/types/data_context_variables.py

```diff
@@ -0,0 +1,111 @@
+import enum
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Any, Optional
+
+import requests
+
+
+class VariableSchema(enum.Enum):
+    CONFIG_VERSION = "config_version"
+    DATASOURCES = "datasources"
+    EXPECTATIONS_STORE_NAME = "expectations_store_name"
+    VALIDATIONS_STORE_NAME = "validations_store_name"
+    EVALUATION_PARAMETER_STORE_NAME = "evaluation_parameter_store_name"
+    CHECKPOINT_STORE_NAME = "checkpoint_store_name"
+    PROFILER_STORE_NAME = "profiler_store_name"
+    PLUGINS_DIRECTORY = "plugins_directory"
+    VALIDATION_OPERATORS = "validation_operators"
+    STORES = "stores"
+    DATA_DOCS_SITES = "data_docs_sites"
+    NOTEBOOKS = "notebooks"
+    CONFIG_VARIABLES_FILE_PATH = "config_variables_file_path"
+    ANONYMIZED_USAGE_STATISTICS = "anonymous_usage_statistics"
+    STORE_BACKEND_DEFAULTS = "store_backend_defaults"
+    CONCURRENCY = "concurrency"
+    PROGRESS_BARS = "progress_bars"
+
+
+@dataclass
+class DataContextVariables:
```

Comment:
Base class does most of the heavy lifting - all CRUD operations are defined herein and any side effects relevant to a particular environment (file I/O or HTTP request) are defined by the appropriate child classes.